component/site/views/flat/icals/tmpl/ical_form.php

```diff
@@ -27,7 +27,7 @@
 			continue;
 		$cats[] = $cid;
 	}
-	if (count($cats) == 0)
+	if (!empty($cats) && count($cats) == 0)
```

Comment:
See https://www.php.net/manual/en/function.empty.php if $cats is array() then empty returns true.  So your logic cancels out here.

Is the purpose of this change to allow export of all categories?

Reply:
set to is_array